src/models/grouping.interface.ts

```diff
@@ -3,7 +3,7 @@ import { GroupingComparerItem } from "./groupingComparerItem.interface";
 import { GroupingFormatterItem } from "./groupingFormatterItem.interface";
 import { SortDirectionNumber } from "./sortDirectionNumber.enum";
 
-export type GroupingGetterFunction<T = any> = (value: T) => string;
+export type GroupingGetterFunction<T = any> = (value: T) => any;
```

Comment:
now thinking about it and I think it should actually be returning type `T` when the Generic Type is provided (which defaults to `any` when not provided), it is for the value so that should be the correct type to use. Sorry for asking to change it so many time 😅

Reply:
damn I think I'm wrong again and this is the value not the datacontext, so it could really be `any`thing